CSRF protection middleware

k0ka opened the pull request:
- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

Resolves #1582

**Changes**

Added the `EnforceJsonContentType` as we discussed in the issue above.

**Breaking changes**

None. The middleware is not enabled by default.

spawnia reviewed (changes requested):
I think a broader set of content types should be valid, see https://github.com/laragraph/utils/blob/master/src/RequestParser.php
- `application/graphql`
- `application/graphql+json`

k0ka commented:
I've rewritten the middleware to make it more permissive and general. Now it allows any request that is not `GET` or `form POST`. And also renamed it to make it more clear.

spawnia commented:
Thanks, nicely done
